### PR TITLE
Add circleci and dockerhub configuration

### PR DESCRIPTION
**Pull request by milescrabill**
@relud r?

### CODE REVIEW

**Comment by milescrabill** [comment]
I will rebase this before merging.

File: Makefile [hunk 1]
```diff
@@ -8,12 +8,12 @@ run:
 		-p 9009:9009 \
 		--env ASR_HOST=localhost \
 		--env ASR_PORT=9009 \
-		-v $(shell pwd):/app/ \
+    -v $(shell pwd):/app/ \
```
Comment (relud) [inline Makefile]: fix yr indent

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+      - run:
+          name: Run tests
+          command: sleep 5 && make test
+
+      - run:
+          name: Push to Dockerhub
+          command: |
+            # set DOCKER_DEPLOY=true in Circle UI to push to Dockerhub
+            DOCKER_DEPLOY="${DOCKER_DEPLOY:-false}"
```
Comment (relud) [inline .circleci/config.yml]: export this, or it's not always available to the deploy-dockerhub.sh script

File: Makefile [hunk 3]
```diff
@@ -1,24 +1,23 @@
 
 build:
-	docker build -t app:build .
+  docker build -t app:build .
```
Comment (relud) [inline Makefile]: why are you converting the tab indents in this file to space indents?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      - run:
+          name: Create version.json
+          command: |
+            # create a version.json per https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
+            printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
+            "$CIRCLE_SHA1" \
+            "$CIRCLE_TAG" \
+            "$CIRCLE_PROJECT_USERNAME" \
+            "$CIRCLE_PROJECT_REPONAME" \
+            "$CIRCLE_BUILD_URL" > ~/speaktome/version.json
```
Comment (relud) [inline .circleci/config.yml]: why are you referencing this file using the full path instead of putting it in . (since docker is built on .)

**Pull request merged**